libs/MeshKernel/include/MeshKernel/Mesh2DToCurvilinear.hpp

```diff
@@ -52,6 +52,129 @@ namespace meshkernel
         std::unique_ptr<CurvilinearGrid> Compute(const Point& point);
 
     private:
+        /// @brief Matrix class supporting negative indices, mimicking behavior similar to Fortran.
+        /// This class allows matrix access with negative indices, which is not possible in C++ by default.
+        class MatrixWithNegativeIndices
+        {
+        public:
+            /// @brief Retrieves the value at the given position (j, i) in the matrix.
+            /// @param j Row index, can be negative.
+            /// @param i Column index, can be negative.
+            /// @return Value at the specified position.
+            int getValue(int j, int i) const
+            {
+                const auto jIndex = j - m_minJ;
+                const auto iIndex = i - m_minI;
+                const auto value = m_matrix(jIndex, iIndex);
+                return value;
+            }
+            /// @brief Sets the value at the given position (j, i) in the matrix.
+            /// @param j Row index, can be negative.
+            /// @param i Column index, can be negative.
+            /// @param value Value to be set at the specified position.
+            void setValue(int j, int i, int value)
+            {
+                const auto jIndex = j - m_minJ;
+                const auto iIndex = i - m_minI;
+                m_matrix(jIndex, iIndex) = value;
+            }
+
+            /// @brief Checks if the position (j, i) in the matrix is valid (not equal to missing value).
+            /// @param j Row index, can be negative.
+            /// @param i Column index, can be negative.
+            /// @return True if the value at the position is not missing, otherwise false.
+            bool IsValid(int j, int i) const
+            {
+                const auto jIndex = j - m_minJ;
+                const auto iIndex = i - m_minI;
+                return m_matrix(jIndex, iIndex) != missing::intValue;
+            }
+
+            /// @brief Gets the number of rows in the matrix.
+            /// @return Number of rows.
+            [[nodiscard]] int rows() const
+            {
+                return static_cast<int>(m_matrix.rows());
+            }
+
+            /// @brief Gets the number of columns in the matrix.
+            /// @return Number of columns.
+            [[nodiscard]] int cols() const
+            {
+                return static_cast<int>(m_matrix.cols());
+            }
+
+            /// @brief Gets the minimum column index (i) allowed in the matrix.
+            /// @return Minimum column index.
+            [[nodiscard]] int minI() const
+            {
```

Comment:
Do you think this (and the other minJ, maxI -J) functions should be renamed to, something like, rowMin (or minRow).